IoTHubBuddy/IoTViewer/AccountManager.cs

```diff
@@ -0,0 +1,362 @@
+
+        const string AccountsContainer = "accountssettingscontainer";
+        const string ProviderIdSubContainer = "providers";
+        const string AuthoritySubContainer = "authorities";
+
+        const string MicrosoftProviderId = "https://login.microsoft.com";
+        const string MicrosoftAccountAuthority = "consumers";
+        const string AzureActiveDirectoryAuthority = "organizations";
+
+        const string MicrosoftAccountClientId = "none";
+        const string MicrosoftAccountScopeRequested = "wl.basic";
+        const string AzureActiveDirectoryClientId = "5cec90d8-14eb-41ac-9b2c-b3e8a7d95da5";
+        const string AzureActiveDirectoryScopeRequested = "wl.basic";
+        private static MainPage mp;
+        /// <summary>
+        /// build the account settings pane with the cached accounts
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        /// <param name="page"></param>
+        public static async void OnAccountCommandsRequested(
+           AccountsSettingsPane sender,
+           AccountsSettingsPaneCommandsRequestedEventArgs e, MainPage page)
+        {
+            mp = page;
+            // In order to make async calls within this callback, the deferral object is needed
+            AccountsSettingsPaneEventDeferral deferral = e.GetDeferral();
+            CreateLocalDataContainers();
+            await AddWebAccountProvidersToPane(e);
+            await AddWebAccountsToPane(e);
+
+            deferral.Complete();
+        }
+        /// <summary>
+        /// find all providers and add to pane
+        /// </summary>
+        /// <param name="e"></param>
+        /// <returns></returns>
+        private static async Task AddWebAccountProvidersToPane(AccountsSettingsPaneCommandsRequestedEventArgs e)
+        {
+            List<WebAccountProvider> providers = await GetAllProviders();
+
+            foreach (WebAccountProvider provider in providers)
+            {
+                WebAccountProviderCommand providerCommand = new WebAccountProviderCommand(provider, WebAccountProviderCommandInvoked);
+                e.WebAccountProviderCommands.Add(providerCommand);
+            }
+        }
+        /// <summary>
+        /// find all cached accounts and add to pane
+        /// </summary>
+        /// <param name="e"></param>
+        /// <returns></returns>
+        private static async Task AddWebAccountsToPane(AccountsSettingsPaneCommandsRequestedEventArgs e)
+        {
+            List<WebAccount> accounts = await GetAllAccounts();
+
+            foreach (WebAccount account in accounts)
+            {
+                WebAccountCommand command = new WebAccountCommand(account, WebAccountInvoked, SupportedWebAccountActions.Remove);
+                e.WebAccountCommands.Add(command);
+            }
+
+        }
+        /// <summary>
+        /// Get a list of stored accounts
+        /// NOTE: the storage model is old, and will be changed soon
```

Comment:
Nope. The storage system in place right now is from the WAM tutorial. If I can get WAM to work, I was planning on looking into the APIs you sent me about storage account info and changing it.